napari/layers/surface/surface.py

```diff
@@ -27,22 +27,41 @@ class Surface(IntensityVisualizationMixin, Layer):
     """
     Surface layer renders meshes onto the canvas.
 
+    Note that surfaces may be colored by:
+        * texture (requires both texture and texcoords to be set)
+        AND
+            * vertex_colors (higher precedence)
+            OR
+            * vertex_values + colormap
+
     Parameters
     ----------
     data : 2-tuple or 3-tuple of array
         The first element of the tuple is an (N, D) array of vertices of
-        mesh triangles. The second is an (M, 3) array of int of indices
-        of the mesh triangles. The optional third element is the
-        (K0, ..., KL, N) array of values used to color vertices where the
-        additional L dimensions are used to color the same mesh with
-        different values. If not provided, it defaults to ones.
+        mesh triangles.
+
+        The second is an (M, 3) array of int of indices of the mesh triangles.
+
+        The optional third element is the (K0, ..., KL, N) array of values
+        (vertex_values) used to color vertices where the additional L
+        dimensions are used to color the same mesh with different values. If
+        not provided, it defaults to ones.
     colormap : str, napari.utils.Colormap, tuple, dict
         Colormap to use for luminance images. If a string must be the name
         of a supported colormap from vispy or matplotlib. If a tuple the
         first value must be a string to assign as a name to a colormap and
         the second item must be a Colormap. If a dict the key must be a
         string to assign as a name to a colormap and the value must be a
         Colormap.
+    texture: (I, J) or (I, J, C) array
+        A 2D texture to be mapped onto the surface.
+        C may be 3 (RGB) or 4 (RGBA) channels for a color texture.
+    texcoords: (N, 2) array
+        2D coordinates for each vertex, mapping into the texture.
```

Comment:
specify the coordinate space here?

Reply:
I updated this but didn't include anything about what happens if your coordinates are outside this range, because I think it's configurable in OpenGL so the fact they wrap is a vispy implementation detail.

Reply:
That still might be useful info to include in the docstring, but I'm +/-1 about whether it's TMI. 😂